NumberFormatException thrown when executing graph query that returns infinity

### Expected behavior

I would expect the query to run without any problem.

### Actual behavior
An exception is thrown:

```
Caused by: java.lang.NumberFormatException: For input string: "inf"
	at java.base/jdk.internal.math.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2054)
	at java.base/jdk.internal.math.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
	at java.base/java.lang.Double.parseDouble(Double.java:543)
	at redis.clients.jedis.graph.ResultSetBuilder.deserializeScalar(ResultSetBuilder.java:259)
	at redis.clients.jedis.graph.ResultSetBuilder.parseRecords(ResultSetBuilder.java:155)
	at redis.clients.jedis.graph.ResultSetBuilder.build(ResultSetBuilder.java:62)
	at redis.clients.jedis.graph.ResultSetBuilder.build(ResultSetBuilder.java:23)
	at redis.clients.jedis.Response.build(Response.java:62)
	at redis.clients.jedis.Response.get(Response.java:38)
	at redis.clients.jedis.TransactionBase.exec(TransactionBase.java:141)
	at redis.clients.jedis.Transaction.exec(Transaction.java:47)
```

### Steps to reproduce:

Run the following to reproduce the exception:
```java
try (Jedis resource = pool.getResource()) {
    try (Transaction transaction = new Transaction(resource)) {
        transaction.graphQuery("db", "RETURN 10^100000");
        transaction.exec();
    }
}
```

RedisGraph returns `inf` / `-inf` which cannot be parsed by `Double.parseDouble`. Similarly, the same thing happens with `NaN` values.

### Redis / Jedis Configuration
**Jedis version**: 4.2.3
**Redis version**: 6.2.6
**Java version**: 11

--- a/src/main/java/redis/clients/jedis/graph/ResultSetBuilder.java
+++ b/src/main/java/redis/clients/jedis/graph/ResultSetBuilder.java
@@ -256,7 +256,7 @@ class ResultSetBuilder extends Builder<ResultSet> {
       case BOOLEAN:
         return Boolean.parseBoolean(SafeEncoder.encode((byte[]) obj));
       case DOUBLE:
-        return Double.parseDouble(SafeEncoder.encode((byte[]) obj));
+        return BuilderFactory.DOUBLE.build(obj);
       case INTEGER:
         return (Long) obj;
       case STRING:

--- a/src/test/java/redis/clients/jedis/modules/graph/GraphValuesTest.java
+++ b/src/test/java/redis/clients/jedis/modules/graph/GraphValuesTest.java
@@ -1,0 +1,25 @@
+package redis.clients.jedis.modules.graph;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import redis.clients.jedis.graph.Record;
+import redis.clients.jedis.graph.ResultSet;
+import redis.clients.jedis.modules.RedisModuleCommandsTestBase;
+
+public class GraphValuesTest extends RedisModuleCommandsTestBase {
+
+  @BeforeClass
+  public static void prepare() {
+    RedisModuleCommandsTestBase.prepare();
+  }
+
+  @Test
+  public void parseInfinity() {
+    ResultSet rs = client.graphQuery("db", "RETURN 10^100000");
+    assertEquals(1, rs.size());
+    Record r = rs.iterator().next();
+    assertEquals(Double.POSITIVE_INFINITY, r.getValue(0), 0d);
+  }
+}

